core/src/main/java/org/apache/carbondata/core/metadata/datatype/DecimalConverterFactory.java

```diff
@@ -173,6 +221,23 @@ public int getSizeInBytes() {
       return new BigDecimal(bigInteger, scale);
     }
 
+    @Override public void fillVector(Object valuesToBeConverted, int size, ColumnVectorInfo info,
+        BitSet nullBitset) {
+      CarbonColumnVector vector = info.vector;
+      int precision = info.measure.getMeasure().getPrecision();
+      if (valuesToBeConverted instanceof byte[][]) {
+        byte[][] data = (byte[][]) valuesToBeConverted;
+        for (int i = 0; i < size; i++) {
+          if (nullBitset.get(i)) {
+            vector.putNull(i);
+          } else {
+            BigInteger bigInteger = new BigInteger(data[i]);
+            vector.putDecimal(i, new BigDecimal(bigInteger, scale), precision);
```

Comment:
here can we use the code `vector.putDecimal(i, DataTypeUtil.byteToBigDecimal(data[i]), precision)`. This is same as used i below fillVector method. If it can be used then we can refactor the code and create only 1 method

Reply:
The method `DataTypeUtil.byteToBigDecimal(data[i]), precision)` is different as it calculates scale also from binary, but here we no need of it.